packages/pigeon/mock_handler_tester/test/test.dart

```diff
@@ -46,6 +46,8 @@ class _TestHostApiCodec extends StandardMessageCodec {
 ///
 /// This comment also tests multiple line comments.
 abstract class TestHostApi {
+  static TestDefaultBinaryMessengerBinding? get _testBinaryMessengerBinding =>
```

Comment:
Why is this being written once per class instead of at the file level?

Reply:
Main motivation was to keep the generator script more readable by generating the `_testBinaryMessengerBinding` getter close to where it is used and to make sure that the public `DartGenerator.writeFlutterApi` always generated functioning code - no matter from what context it is called. If we want to generate this only once per file, we'd have to move the generation out of `DartGenerator.writeFlutterApi` into `DartGenerator.generateTest` - it's possible. If you prefer that, let me know.

Reply:
I'm not as familiar with the test generator; that seems reasonable, so feel free to leave it.

(Long term we want to do more to have shared utilities created in a clear, structured way, but that work may not have even started for this generator.)